code/game/machinery/_machinery.dm

```diff
@@ -159,10 +159,18 @@
 	fire = 50
 	acid = 70
 
-/obj/machinery/Initialize(mapload)
+/obj/machinery/Initialize(mapload, obj/structure/frame/machine/frame_with_parts = null, mob/living/builder = null, obj/item/building_tool = null)
```

Comment:
There is no need to change the main machinery Initialize proc when the extra params are only used by the machine frame.

The null  and istype() check adds extra init time to all machines for no reason, revert this